src/core/symbology/qgsstyle.cpp

```diff
@@ -2755,6 +2755,7 @@ bool QgsStyle::importXml( const QString &filename, int sinceVersion )
       if ( entityAddedVersion != 0 && sinceVersion != -1 && entityAddedVersion <= sinceVersion )
       {
         // skip the symbol, should already be present
+        e = e.nextSiblingElement();
```

Comment:
instead of having to add this e = e.nextSiblingElement() at each continue statement, transforming the while() loop into a for() would be more appropriate (should be done in master first though)

Reply:
> (should be done in master first though)

or at least, as a follow-up cleanup.